internal/generator/generator_util_test.go

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package generator
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func getProjectRootAbsPath() (string, error) {
+	_, filePath, _, ok := runtime.Caller(0)
+	if !ok {
+		return "", errors.New("can't get path of generator_util_test.go through runtime")
+	}
+	absPath, err := filepath.Abs(filePath)
+	if err != nil {
+		return "", err
+	}
+	// get the project root directory
+	rootDir := filepath.Dir(absPath)
+	rootDir = filepath.Dir(rootDir)
+	rootDir = filepath.Dir(rootDir)
+
+	return rootDir, nil
+}
+
+func getTestSrcCodePath(relativePath string) (string, error) {
+	root, err := getProjectRootAbsPath()
+	if err != nil {
+		return "", err
+	}
+	return path.Join(root, "internal", "generator", "testdata", "source_codes", relativePath), nil
+}
+
+func getExpectedFilePath(relativePath string) (string, error) {
+	root, err := getProjectRootAbsPath()
+	if err != nil {
+		return "", err
+	}
+	relativePath = strings.TrimSuffix(relativePath, ".c")
+	relativePath = strings.TrimSuffix(relativePath, ".cpp")
+	return path.Join(root, "internal", "generator", "testdata", "expected", relativePath), nil
+}
+
+//nolint:gochecknoglobals
+var (
+	update = flag.Bool("update", false,
+		`update the expected output of these tests, 
+only use when the expected output is outdated and you are sure your output is correct`)
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	os.Exit(m.Run())
+}
```

Comment:
Nit: I think it would be nice to move this to the top of the file so it's easier to spot.

Reply:
Sure! Committed